core/primitives/src/lib.rs

```diff
@@ -88,19 +88,36 @@ pub enum ExecutionContext {
 	Syncing,
 	/// Context used for block construction.
 	BlockConstruction,
-	/// Offchain worker context.
+	/// Context used for offchain calls.
+	///
+	/// Includes keystore capabilities.
+	OffchainCall,
+	/// Context used for rich offchain calls.
+	///
+	/// This includes access to offchain worker db (read / write)
+	/// and transaction pool.
+	RichOffchainCall(Box<dyn offchain::Externalities>),
```

Comment:
As we talked on the retreat, I really would like to merge these two calls. (`RichOffchainCall` and `OffchainCall`)

I don't know if you want to do this in this pr? Could also be done by me/you in a later pr.

Reply:
Well, the whole point of this PR is to actually split the two calls. I don't want to allow people to make HTTP calls from the runtime everytime the node calls into some function.
After discussions with Andre and Rob we'd figured out that `RichOffchainCall` is going to be specifically needed for some cases (like equivocation report generation) and I think we should really limit the APIs that are available in such case (i.e. we only want the ones listed on the capabilities list). I think I'll even split read and writes to offchain worker db, so that `RichOffchainCall` is only allowed to read.

Reply:
I like the idea, I would just make `OffchainCall` take the capabilities and make them by default allow nothing or so. I don't see why we want to maintain two versions. But we can do this later, when I refactor the keystore usage.

Reply:
As discussed "offchain", we will revisit this later.